ddtrace/ddtrace.go

```diff
@@ -79,6 +79,48 @@ type SpanContext interface {
 	ForeachBaggageItem(handler func(k, v string) bool)
 }
 
+// ReadableSpan specifies methods to read from a span.
+type ReadableSpan interface {
+	// GetName returns the span's operation name.
+	GetName() string
+
+	// GetService returns the span's service name.
+	GetService() string
+
+	// GetResource returns the span's resource name.
+	GetResource() string
+
+	// GetType returns the span's type.
+	GetType() string
```

Comment:
I would also remove these, and retrieve them only as spans. It is always easy to add them later on if needed, but it will be impossible to remove them once added.